### ISSUE TITLE
Can't upgrade from 2.0.3 to 2.1.

### ISSUE THREAD

**Issue by roxy-dao** [msupply-foundation/open-msupply#4238]
## What went wrong? 😲
```
    0: Error during migration (2.1.0)
    1: Sql error 
               CREATE TABLE store_new (
                 id TEXT NOT NULL PRIMARY KEY,
                 name_link_id TEXT NOT NULL REFERENCES name_link(id),
                 code TEXT NOT NULL,
                 site_id INTEGER NOT NULL,
                 logo TEXT,
                 store_mode TEXT DEFAULT 'STORE' NOT NULL,
                 created_date TEXT
               );
       
               INSERT INTO store_new SELECT * FROM store;
       
               PRAGMA foreign_keys=off;
               DROP TABLE store;
               ALTER TABLE store_new RENAME TO store;
               PRAGMA foreign_keys=on;
            
    2: "UNKNOWN" ("\"\\\"NOT NULL constraint failed: store_new.store_mode\\\"\"")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

## How to Reproduce 🔨

Steps to reproduce the behaviour:

## Your environment 🌱

- Open mSupply Version:
- Legacy mSupply Central Server Version:

- Platform:

- Database type:

**Comment by roxy-dao** [comment]
Closed in #4241

**Comment by marthakeezy** [comment]
Might be a bit irrelevant now, but tested on apk -> upgrading from 2.0.3 to 2.1 and all works well. Moving to Done ✅
